tools/autograd/load_derivatives.py

```diff
@@ -129,6 +129,54 @@ def set_up_derivatives(defn, declaration):
                 args.append(name)
             derivatives.append(create_derivative(declaration, formula, output_indices, args))
 
+        # Test to see if the use of 'grads' makes sense.
+        used_grad = 0
```

Comment:
This is a large chunk of the outer function, can we move it out somewhere else with an appropriate name, so we don't make this one much less readable?